Потому что Exception обрабатывать обязательно,
а RuntimeException не обязательно.